### ISSUE TITLE
Make it possible to specify the full TargetGroup name when using an ALB event

### ISSUE THREAD

**Issue by cbm-gplassard** [serverless/serverless#9213]
### Use case description

Hello, we would like to make the TargetGroup names more predictable / human readable when using AWS Lambda with ALB. There are several reasons for this but mostly because we would like to make the CloudWatch Metrics easier to use (when displayed on dashboards, when creating alarms, ...).

From what I understand today the name is created with a hash, even though it's possible to specify a common prefix. The prefix would not be enough for our use case since it is defined at the provider level.

Would it be possible to add a parameter to specify the full TargetGroup name when creating the event ?  It would then be the user's responsibility to ensure there are no duplicates. I'm not exaclty sure how it should behave in conjonction with the targetGroupPrefix.

### Proposed solution

```yml
functions:
  albEventConsumer:
    handler: handler.hello
    events:
      - alb:
          listenerArn: arn:aws:elasticloadbalancing:us-east-1:12345:listener/app/my-load-balancer/50dc6c495c0c9188/
          priority: 1
          targetGroupName: albEventConsumer-${opt:stage}
          conditions:
            path: /hello
```

Thank you !

**Comment by pgrzesik** [comment]
Hello @cbm-gplassard :wave: Thanks for submitting your idea for an improvement. I believe the proposal you presented looks really solid. At the moment, we're using a generated name: https://github.com/serverless/serverless/blob/5e16a28a5165d3547ed9f0b6d77114c7abb7ef21/lib/plugins/aws/package/compile/events/alb/lib/targetGroups.js#L65 so we would have to introduce logic that will not generate a new name if there's one already provided. Additionally, I think that we might need to adjust the "Name" tag generated and assigned here: https://github.com/serverless/serverless/blob/5e16a28a5165d3547ed9f0b6d77114c7abb7ef21/lib/plugins/aws/package/compile/events/alb/lib/targetGroups.js#L69

We'd be more than happy to accept a PR that introduces this improvement :raised_hands:

**Comment by cbm-gplassard** [comment]
Hello @pgrzesik 
Thank you I will work on a PR !

**Comment by pgrzesik** [comment]
That's an awesome news @cbm-gplassard :raised_hands: Feel free to reach out whenever you have any questions :100:

**Comment by medikoo** [comment]
Addressed by #9222
